File: platform/Leetcode/src/reverse_letters.py
def reverse_letters(string):
    phrase = list(string)
    indices, letters = [], []
    
    for ind,letter in enumerate(string):
        if letter.isalpha():
            indices.append(ind)
            letters.append(letter)
    
    reversed_indices = indices[::-1]
    for i,letter in zip(reversed_indices, letters):
        phrase[i] = letter
    
    return ''.join(phrase)

def main():
    user_input = input("Enter a string of ASCII codes (e.g., alphabets, punctuations, etc.): ")
    output = reverse_letters(user_input)
    print("Reversing only the alphabets in the string, we have: ", output)
    
if __name__ == '__main__':
    main()

File: platform/Leetcode/src/ReverseLetters.java
import java.util.Arrays;
import java.util.ArrayList;
import java.util.Scanner;

public class ReverseLetters{

    public static void main(String[] args){
        System.out.println("Enter a string of ascii codes (e.g., letters, punctuations, etc.)");
        Scanner sc = new Scanner(System.in);
        String userInput = sc.nextLine();
        String output = reverseLettersOnly(userInput);
        System.out.println("When ONLY the letters are reversed, you'll get: " + output);
            
    }
    public static String reverseLettersOnly(String word){
        char[] phrase = word.toCharArray();

        ArrayList<Integer> indices = new ArrayList<Integer>();
        ArrayList<Character> letters = new ArrayList<Character>();

        for (int i=0; i < phrase.length; i++) {
            if (Character.isLetter(phrase[i])) {
                indices.add(i);
                letters.add(phrase[i]);
            }
        }

        ArrayList<Integer> reversedIndices = new ArrayList<Integer>();
        for (int i=indices.size()-1; i>=0 ; i--) {
            reversedIndices.add(indices.get(i));
        }

        for (int index:reversedIndices) {
            phrase[index] = letters.remove(0);        
        }

        return String.join("",String.copyValueOf(phrase));
    }
}
